### DISCUSSION TITLE
Shortcut to start the GUI

### DISCUSSION THREAD

**Discussion by healthyfat** [bmaltais/kohya_ss#50] [Ideas]
Create new file inside /kohya_ss folder with *.bat extension (eg. gui.bat) and paste below code: 

> @echo off
> 
> set VENV_DIR=.\venv
> set PYTHON=python
> 
> call %VENV_DIR%\Scripts\activate.bat
> 
> %PYTHON% kohya_gui.py --inbrowser
> 
> pause

or 

> @echo off
> 
> call venv\Scripts\activate.bat
> python.exe kohya_gui.py --inbrowser %*
> 
> pause

Not sure what is the difference, but both ways work. 

Now you can start the GUI with one click. 

EDIT:

Add this argument to autolaunch GUI in browser. Suggested by @txirrindulari 

%PYTHON% kohya_gui.py --inbrowser

**Comment by bmaltais** [comment]
I will add the bat file to the repo. It should help other users... should be included in the next release. Thank's for the submission!

**Reply by dpons222** [reply]
Hey! I just downloaded Kohya, the user bat file with "--inbrowser" is not included anymore? (sorry, kinda new)

**Comment by Pico52** [comment]
> If someone knows the trick to include URL autolaunch that would be neat.

I believe what you are referring to is:
`start "" "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" "--incognito 127.0.0.1:7860"`

If you don't use Chrome, obviously you would just change the path to whatever it is you use.  I always use incognito mode for A1111 to avoid any extension compatibility issues.

I also want to add that if you're a Windows 10 user and would like to run the server in a Windows Terminal instead of the standard Command Prompt, you can change your registry to execute batch files in a Windows Terminal instead.

Windows Key > Type regedit > Navigate to: Computer\HKEY_CLASSES_ROOT\batfile\shell\open\command > Set the "(Default)" Value Data to: `wt.exe -w 0 new-tab -d . "%1" %*`

This should execute the wt.exe (Windows Terminal) on the first available instance (-w 0) in a new tab.  This should execute from the directory from where it was called.

I was really hoping someone would have an icon made though, which is why I came to this thread.  I like to send a shortcut of my batch launchers to my desktop and change their icons to something appropriate.

**Reply by healthyfat** [reply]
> I believe what you are referring to is:
> start "" "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" "--incognito 127.0.0.1:7860"

That doesn't work.

**Comment by vivi-gomez** [comment]
Add --inbrowser in your .bat file and it will automatically open the browser
%PYTHON% kohya_gui.py --inbrowser
And if you want to change default port 7860 to something else, 7890 for example 
--server_port 7890

**Reply by healthyfat** [reply]
That works perfect. Thank you. 

How did you find out this argument?

**Reply by vivi-gomez** [reply]
bmaltais just added these options a few days ago, and I was reading and learning in the repo
https://github.com/bmaltais/kohya_ss/pull/138#issue-1578803738
